### DISCUSSION TITLE
A general question on licensing scripts using biomod2

### DISCUSSION THREAD

**Discussion by Farewe** [biomodhub/biomod2#442] [Q&A] [answered]
Hello biomod team 

I have a general question on the licence used by biomod2 and how you see derivative works. The biomod2 licence is GPLv3 meaning that derivative works should also apply a GPLv3 licence. However, it is unclear to me whether an R script that uses biomod2 should be considered as a derivative work or as a script interacting with biomod2. I asked chatgpt if there is a clear answer to that question or not (for what it's worth): 

> "If your script merely calls or uses the R package without incorporating its code, the situation might be less clear-cut. The GPL's requirements mainly concern distribution of derivative works. The nature of R scripts (often interpreted rather than compiled) and the way R packages are used (often as independent but required components) can blur the lines between derivative and separate works."

Maybe having your take on this as developers of the package and owners of the licence would be better: under what license would you prefer scripts using biomod2 to be distributed?

I am hesitating between GPLv3 and MIT licences. 

Thanks a lot 
Boris

**Comment by MayaGueguen** [comment, marked as answer]
Hello Boris,

I might be wrong as I am not a specialist on the licence subject.
But as you said, `GPLv3` implies that derivative works should also apply a `GPLv3` licence, mostly meaning that people modifying code should provide back this code to the community and that your part of code (if you added one) should also be provided, while this it not required by the `MIT` licence.
This covers the fact that if the code is integrated into another software, and then modified, developers have to say that it has been modified.
`GPLv3` also prevents the code from being sold with additional modifications under a commercial licence.
So as you said, it is not crystal clear how much an R script calling `biomod2` is derivative work, but with the idea of `biomod2` staying in the public world of research, we would prefer to keep the continuity with the `GPLv3` licence.

Maya

**Reply by Farewe** [reply]
Yes, I agree with you, it's not clear at all how licenses interact among R packages and their dependencies. I think this is overlooked by many when publishing scripts or new R packages.
For example on the CRAN three packages are importing `biomod2`, and two of them use a `GPLv2` license (e.g. [here](https://cran.r-project.org/web/packages/gecko/index.html)) which is less strict than the `GPLv3` license. They should probably apply the `GPLv3` license as well under the assumption that importing a `GPLv3` dependency requires to apply a `GPLv3` license to the whole code. 

I am still not clear if it should be the case or not, but at least I have your opinion on the license you would like to apply. I will likely apply a `GPLv3` license I distribute my scripts that rely on biomod2.
